test/OrchardCore.Commerce.Tests.UI/Tests/InventoryTests/InventoryBehaviourTests.cs

```diff
@@ -0,0 +1,122 @@
+using Atata;
+using Lombiq.Tests.UI.Attributes;
+using Lombiq.Tests.UI.Extensions;
+using Lombiq.Tests.UI.Services;
+using OpenQA.Selenium;
+using System.Globalization;
+using Xunit;
+using Xunit.Abstractions;
+using static OrchardCore.Commerce.Tests.UI.Constants.ContentItemIds;
+
+namespace OrchardCore.Commerce.Tests.UI.Tests.PromotionTests;
+
+public class InventoryBehaviourTests : UITestBase
+{
+    public InventoryBehaviourTests(ITestOutputHelper testOutputHelper)
+        : base(testOutputHelper)
+    {
+    }
+
+    [Theory, Chrome]
+    public Task InventoryChecksOnCartUpdateShouldWorkProperly(Browser browser) =>
+        ExecuteTestAfterSetupAsync(
+            async context =>
+            {
+                await context.SignInDirectlyAndGoToRelativeUrlAsync("/testproduct");
+                await context.ClickReliablyOnAsync(By.XPath("//button[contains(., 'Add to cart')]"));
+
+                // When quantity is set to a value below the possible max quantity, error message should not appear.
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "2",
+                    "The checkout quantity for Test Product is more than the maximum allowed (2).",
+                    shouldExist: false);
+
+                // When quantity is set to a value above the maximum possible quantity, error message should appear.
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "5",
+                    "The checkout quantity for Test Product is more than the maximum allowed (2).",
+                    shouldExist: true);
+
+                await context.GoToAdminRelativeUrlAsync($"/Contents/ContentItems/{TestProduct}/Edit");
+                await context.ClickReliablyOnAsync(By.XPath("//input[@id='InventoryPart_AllowsBackOrder_Value']"));
```

Comment:
In HTML `id` values have to be unique (this is automatically verified by `html-validate` during UI testing) so there is no point checking the tag name too. You can simplify to `By.Id("InventoryPart_AllowsBackOrder_Value")` instead. Same below.